src/main/java/org/cloudfoundry/community/servicebroker/s3/service/iam/Iam.java

```diff
@@ -0,0 +1,142 @@
+package org.cloudfoundry.community.servicebroker.s3.service.iam;
+
+import com.amazonaws.services.identitymanagement.AmazonIdentityManagement;
+import com.amazonaws.services.identitymanagement.model.*;
+import org.cloudfoundry.community.servicebroker.s3.policy.BucketGroupPolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+public abstract class Iam {
```

Comment:
To make this approach more clear, we've split up https://github.com/mendix/s3-cf-service-broker/commit/9378a54 from the old pull request into:
- https://github.com/mendix/s3-cf-service-broker/commit/cf4a4e7 [create a Plan interface and BasicPlan implementation to support multiple plans]
- https://github.com/mendix/s3-cf-service-broker/commit/f175b9c [split BasicPlanIam from Iam to support multiple plans]

In the 2nd commit we made ```Iam.java``` a more generic class and introduced ```BasicPlanIam.java```. ```getUserNameForBinding()```, ```getGroupNameForInstance()``` and ```getPolicyNameForInstance()``` moved there. The arguments of the functions in ```Iam.java``` don't contain ```bindingId``` or ```instanceId``` anymore. This logic also moved to ```BasicPlanIam.java```.